### ISSUE TITLE
Project with multiple sources do not correctly get their source version mapped

### ISSUE THREAD

**Issue by bnico99** [PolyJIT/benchbuild#524]
In a project with multiple sources, the execution instantly terminates with the message "Could not find any unprocessed variants."  although there are multiple unprocessed variants.

According to @vulder, the issue seems to be related to the ordering of the values for the sources.

For example, for the following sources

```
SOURCE = [
    PaperConfigSpecificGit(
        project_name="clasp",
        remote="https://github.com/potassco/clasp.git",
        local="clasp",
        refspec="origin/HEAD",
        limit=None,
        shallow=False
    ),
    FeatureSource(),
    HTTP(
        local="SAT_H_instances_childsnack_p08.hddl_2.cnf",
        remote={
            "1.0":
                "https://github.com/bnico99/picoSAT-mirror/releases/"
                "download/picoSAT-965/SAT_H_instances_childsnack_p08.hddl_2.cnf"
        }
    )
]
```
the following list of projects is generated for the slurm script:

```
# List of available projects
projects=(
'clasp-cpp_projects@4c708a73fd,1.0,0'
'clasp-cpp_projects@4c708a73fd,1.0,1'
'clasp-cpp_projects@6382096f5a,1.0,0'
'clasp-cpp_projects@6382096f5a,1.0,1'
)
```

**Comment by vulder** [comment]
The unprocessed variants does not have anything to do with BB itself that's on our side. However, the underlying problem is related to the wrong mapping of values to the corresponding sources for a project.

**Comment by simbuerg** [comment]
Thanks for the report. Should be fixed with PR #525

**Comment by bnico99** [comment]
> Thanks for the report. Should be fixed with PR #525

I can confirm that it fixes the issue (at least for my use case). Thanks :)
